pallets/did/src/did_details.rs

```diff
@@ -220,7 +221,9 @@ impl DidVerifiableIdentifier for kilt_primitives::DidIdentifier {
 ///
 /// It is currently used to keep track of all the past and current
 /// attestation keys a DID might control.
-#[derive(Clone, Debug, Decode, Encode, PartialEq, Ord, PartialOrd, Eq)]
+#[derive(Clone, Debug, Decode, Encode, PartialEq, Ord, PartialOrd, Eq, TypeInfo)]
+#[scale_info(skip_type_params(T))]
```

Comment:
Thank you for this helpful comment https://gist.github.com/ascjones/0d81a4c44e84cacd9f714cd34a6de823#gistcomment-3931135

I noticed you described it as `structs which lazily use the pallet's Config as generic`. Do you mean that the less lazy fix would be
```rust
pub struct DidPublicKeyDetails<DidPublicKey> {
	/// A public key the DID controls.
	pub key: DidPublicKey,
}
```

Reply:
Not sure the _lazy_ is the correct technical term but it meant to reflect that instead of specifying the different Generics, the Config trait is provider instead.